src/python/pants/source/source_root.py

```diff
@@ -20,6 +21,14 @@ class SourceRootCategories:
 SourceRoot = namedtuple('_SourceRoot', ['path', 'langs', 'category'])
 
 
+class UncanonicalizedSourceRoot(NamedTuple):
```

Comment:
I mean, at this point we may as well make it a `@dataclass`.

Reply:
If we could do that in 3.6 (which we can via backport), I’d agree. They are very similar and I usually default to dataclass.

Reply:
Oh, are we not 3.7-ing yet? Never mind. We'll wait.

Reply:
I’d love to drop 3.6 so that we could use dataclasses and `from __future__ import annotations`. Twitter just switched to using 3.7, but I think we have several users like Foursquare still on 3.6 and in general we’re planning on supporting the two (three?) most recent Python versions.